$OpenBSD: patch-program_lib_Roundcube_rcube_imap_generic_php,v 1.3 2015/10/21 16:48:50 sthen Exp $

PHP's STREAM_CRYPTO_METHOD_TLS_CLIENT is TLSv1.0 only "for BC with pre-5.6".
There's a STREAM_CRYPTO_METHOD_TLS_ANY_CLIENT but it doesn't work, so use
STREAM_CRYPTO_METHOD_SSLv23_CLIENT which (at least with libressl) is TLS-only
anyway.

Thanks PHP.

--- program/lib/Roundcube/rcube_imap_generic.php.orig	Wed Oct 21 17:34:36 2015
+++ program/lib/Roundcube/rcube_imap_generic.php	Wed Oct 21 17:35:50 2015
@@ -910,7 +910,7 @@ class rcube_imap_generic
                 return false;
             }
 
-            if (!stream_socket_enable_crypto($this->fp, true, STREAM_CRYPTO_METHOD_TLS_CLIENT)) {
+            if (!stream_socket_enable_crypto($this->fp, true, STREAM_CRYPTO_METHOD_SSLv23_CLIENT)) {
                 $this->setError(self::ERROR_BAD, "Unable to negotiate TLS");
                 $this->closeConnection();
                 return false;
